src/controller/CMakeLists.txt

```diff
@@ -12,3 +13,5 @@ include_directories(${MODULE_INCLUDE_DIRS})
 target_include_directories(controller PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
 )
+
+target_link_libraries(controller jsoncpp)
```

Comment:
You need to add jsoncpp as a submodule and link it in the main `CMakeLists.txt`

Reply:
I didn't quite get this. What is the actual issue here?

Reply:
JSON Cpp currently has not been added as a submodule right? I will not have the source for jsoncpp unless it's added as a submodule and cloned into `ext`.

You need to run the right submodule commands to add this to the `.gitmodules` file.

Reply:
Oh, I thought that JsonCpp was a gcc specification and it came with the recent versions of the compiler. I'll add it right away.
I guess this [Jsoncpp](https://github.com/open-source-parsers/jsoncpp.git) is the correct submodule?

Reply:
I'm assuming so, I have no clue how it's working on your machine. Perhaps you have it installed system wide? It might be a dependency for one the installed packages on your system.

Reply:
Yup, makes sense. I'll add it as a `submodule` right away.